File: data/data-jdbc-postgresql/src/main/java/com/example/data/jdbc/postgresql/model/Book.java
package com.example.data.jdbc.postgresql.model;

import java.util.Objects;

import org.springframework.data.annotation.Id;

public class Book {

	@Id
	private final Long id;

	private final String title;

	public Book(Long id, String title) {
		this.id = id;
		this.title = title;
	}

	public Long getId() {
		return id;
	}

	public String getTitle() {
		return title;
	}

	@Override
	public boolean equals(Object o) {
		if (this == o) {
			return true;
		}
		if (o == null || getClass() != o.getClass()) {
			return false;
		}
		Book book = (Book) o;
		return Objects.equals(id, book.id) && Objects.equals(title, book.title);
	}

	@Override
	public int hashCode() {
		return Objects.hash(id, title);
	}

	@Override
	public String toString() {
		return "Book{" + "title='" + title + '\'' + '}';
	}

}

File: data/data-jpa-kotlin/src/main/kotlin/com/example/data/jpa/kotlin/model/Book.kt
/*
 * Copyright 2022-2024 the original author or authors.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.example.data.jpa.kotlin.model

import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.Id
import java.util.*

@Entity
class Book(@Id @GeneratedValue var id: Long?, var title: String?) {

	override fun equals(other: Any?): Boolean {
		if (this === other) {
			return true
		}
		if (other == null) {
			return false
		}
		val book = other as Book
		return id == book.id && title == book.title
	}

	override fun hashCode(): Int {
		return Objects.hash(id, title)
	}

	override fun toString(): String {
		return "Book{title='$title'}"
	}
}
